### ISSUE TITLE
Make mergeSorted support emit eagerly when only one input stream is ready

### ISSUE THREAD

**Issue by hepin1989** [akka/akka#30850]
The currently behavior need to wait both left and right input stream to be ready for emmiting the smallest element. I would expect it can emit the ready one and do comparing only when both are ready.
Refs:https://github.com/reactor/reactor-core/issues/2827

**Comment by johanandren** [comment]
I don't think we should change the existing behavior as it would break existing use but we could potentially explore introducing a flag at construction to tell the stage to emit if only one is available. Note that this would probably only ever sort when downstream back pressures though, so I don't think it would be very intuitive.

**Comment by He-Pin** [comment]
A mergePriority was added in https://github.com/reactor/reactor-core/pull/3267
